tests/test_templates.py

```diff
@@ -12,106 +12,249 @@
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.CVNeuralNetLayer()."""
+cv_func = [(CVNeuralNetLayers, cvqnn_layers_uniform),
+           (CVNeuralNetLayers, cvqnn_layers_normal)]
 
-        p = parfun(n_wires=n_subsystems)
+qubit_const = [(StronglyEntanglingLayers, [[[[4.54, 4.79, 2.98], [4.93, 4.11, 5.58]],
+                                           [[6.08, 5.94, 0.05], [2.44, 5.07, 0.95]]]]),
+               (RandomLayers, [[[0.56, 5.14], [2.21, 4.27]]]),
+               (AmplitudeEmbedding, [[1 / 2, 1 / 2, 1 / 2, 1 / 2]]),
+               (BasisEmbedding, [[1, 0]]),
+               (AngleEmbedding, [[1., 2.]])
+              ]
 
-        @qml.qnode(gaussian_device)
-        def circuit(weights):
-            CVNeuralNetLayer(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+cv_const = [(DisplacementEmbedding, [[1., 2.]]),
+            (SqueezingEmbedding, [[1., 2.]]),
+            (CVNeuralNetLayers, [[[2.31], [1.22]],
+                                 [[3.47], [2.01]],
+                                 [[0.93, 1.58], [5.07, 4.82]],
+                                 [[0.21,  0.12], [-0.09, 0.04]],
+                                 [[4.76, 6.08], [6.09, 6.22]],
+                                 [[4.83], [1.70]],
+                                 [[4.74], [5.39]],
+                                 [[0.88, 0.62], [1.09, 3.02]],
+                                 [[-0.01, -0.05], [0.08, -0.19]],
+                                 [[1.89, 3.59], [1.49, 3.71]],
+                                 [[0.09,  0.03], [-0.14,  0.04]]
+                                 ]),
+            (Interferometer, [[2.31], [3.49], [0.98, 1.54]])
+            ]
 
-        circuit(weights=p)
+#########################################
+# Circuits
 
-    @pytest.mark.parametrize('parfun', [strong_ent_layers_uniform, strong_ent_layers_normal])
-    def test_integration_stronglyentangling_layers(self, parfun, qubit_device, n_subsystems, n_layers):
-        """Checks that parameters generated by methods from pennylane.init integrate
-        with pennnylane.templates.layers.StronglyEntanglingLayers()."""
 
-        p = parfun(n_layers=n_layers, n_wires=n_subsystems)
+def qnode_qubit_args(dev, intrfc, templ1, templ2, n):
+    """QNode for qubit integration circuit using positional arguments"""
+    @qml.qnode(dev, interface=intrfc)
+    def circuit(*inp_):
+        # Split inputs again
+        inp1_ = inp_[:n]
+        inp2_ = inp_[n:]
 
-        @qml.qnode(qubit_device)
-        def circuit(weights):
-            StronglyEntanglingLayers(*weights, wires=range(n_subsystems))
-            return qml.expval(qml.Identity(0))
+        qml.PauliX(wires=0)
+        templ1(*inp1_, wires=range(2))
+        templ2(*inp2_, wires=range(2))
+        qml.PauliX(wires=1)
+        return [qml.expval(qml.Identity(0)), qml.expval(qml.PauliX(1))]
+    return circuit
```

Comment:
Is it worth having a test case for `def circuit(weights):`? (as opposed to `circuit(*weights)`). This is a pattern I attempt to use a lot of the time

Reply:
I will open an issue as a reminder...